hammers/scripts/maintenance_reservation.py

```diff
@@ -1,144 +1,224 @@
+                                    events=[])
+        print("Lease {name} (id: {id}) created successfully!".format(name=lease['name'], id=lease['id']))
+    
+    return start_time_str_in_ct, end_time_str_in_ct
 
+@prometheus_exporter(__file__)
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv
+
+    parser = argparse.ArgumentParser(description='Reserve nodes for maintenance')
+    append_global_identity_args(parser, argv)
+    
+    mysqlargs = MySqlArgs({
+        'user': 'root',
+        'password': '',
+        'host': 'localhost',
+        'port': 3306,
+    })
+    mysqlargs.inject(parser)
+    
+    parser.add_argument('--operator', type=str, required=True, help='Chameleon account username of the operator')
+    parser.add_argument('--nodes', type=str, required=True, help='node ids or node names; comma separated')
+    parser.add_argument('--reason', type=str, required=True, help='maintenance reasons')
+    parser.add_argument('--dry-run', action="store_true", help='perform a trial run without making reservations')
+    parser.add_argument('--start-time', type=valid_date, default=None, help='lease start time in CT (YYYY-mm-DD HH:MM:SS); if not given, start at the earliest possible datetime')
+    parser.add_argument('--estimate-hours', type=int, default=24, help='estimated hours required for maintenance; default is 24 hours')
+    parser.add_argument('--slack', type=str, default=None, help='JSON file with Slack webhook information to send a notification to')
+
+    args = parser.parse_args(argv[1:])
+    
+    # connect to database
+    mysqlargs.extract(args)
+    db = mysqlargs.connect()
+    
+    # keystone authentication
+    auth_args = {'auth_url': args.os_auth_url,
+                 'username': args.os_username,
+                 'password': args.os_password,
+                 'project_id': args.os_project_id,
+                 'project_name': args.os_project_name}
+    if args.os_user_domain_name:
+        auth_args['user_domain_name'] = args.os_user_domain_name
+    if args.os_project_domain_name:
+        auth_args['project_domain_name'] = args.os_project_domain_name
+    # get admin session for node information
+    admin_sess = get_session(**auth_args)
+    # get maint session for creating lease
+    auth_args['project_name'] = 'maintenance'
+    auth_args['project_id'] = '4ffe61cf850d4b45aef86b46411d33e1'
+    maint_sess = get_session(**auth_args)
+    
+    # get node details
+    nodes = get_nodes(admin_sess, args.nodes.split(','))
+    
+    report_info = {}
+    for node in nodes:
+        lease_start_time = args.start_time
+        if not lease_start_time:
+            # find the earliest reservation time for the node
+            lease_start_time = get_node_earliest_reserve_time(db, node.uuid, args.estimate_hours)
+        else:
+            # convert to utc
+            lease_start_time = lease_start_time.replace(tzinfo=tz.gettz('America/Chicago')).astimezone(tz.gettz('UTC'))
```

Comment:
I could be wrong, but I would think that it would automatically parse the timestamp given the timezone set on the system. Not sure what that is set to for our control nodes (or wherever the hammer is run). This might be confusing long-term in any case to default to CT. Perhaps you could specify timezone as a flag?